### ISSUE TITLE
FullyConnectedNet+BatchNormalization

### ISSUE THREAD

**Issue by seokjeongeum** [seungah-Jang/CS182DeepLearning#3]
https://github.com/seungah-Jang/CS182DeepLearning/blob/02a13f5b121dc3a3137313ea2681bf5eabb79b2b/hw1/BatchNormalization.ipynb#L308
작은 에러가 나와야 합니다
https://github.com/seungah-Jang/CS182DeepLearning/blob/02a13f5b121dc3a3137313ea2681bf5eabb79b2b/hw1/deeplearning/classifiers/fc_net.py#L160
구현이 잘못되어 있습니다 
구현의 잘못된 점입니다
Incorrect Initialization of Batch Normalization Parameters
 - The code intends to initialize scale (gamma) and shift (beta) parameters for every hidden layer when batch normalization is enabled. However, the initialization block
  if self.use_batchnorm and i < self.num_layers - 1:
 is placed outside the loop that iterates over layers. Because the loop variable i will be left at self.num_layers – 1 (the last layer) after the loop finishes, the condition i < self.num_layers – 1 is never true. As a result, gamma and beta are not being initialized at all.
 - The fix is to initialize gamma and beta for each hidden layer (i.e., for every layer except the output layer) inside the loop. For example, you could add an inner check inside the loop for i in range(self.num_layers – 1).

Missing Forward/Backward Handling for Dropout and Batch Normalization
 - Although the class is designed to optionally use dropout and batch normalization—as indicated by the parameters and comments—the forward and backward passes do not incorporate those operations. In a complete implementation you would:
  – Include a batch norm forward pass (using the gamma, beta, and bn_param for that layer) between the affine and ReLU steps for each hidden layer.
  – Pass the dropout_param to a dropout forward function (and similarly include a dropout backward pass) when dropout is enabled.
 This omission means that, even if dropout or batch normalization is enabled when constructing the network, those features will have no effect.
